### DISCUSSION TITLE
scribbling canvas with "almost realtime" generation?

### DISCUSSION THREAD

**Discussion by tybiboune** [lllyasviel/Fooocus#2242] [Ideas]
Wouldn't it be awesome to have, inside Fooocus, a new creation tab that would work like this?

https://app.leonardo.ai/realtime-canvas

Adding to this workflow the possibility to use any model & lora and all integrated inside Fooocus.... YUMM :)

**Comment by mashb1t** [comment]
I've tried and implemented that already but image generation is heavily slowed down by the use of the Gradio queue. See https://github.com/lllyasviel/Fooocus/pull/1043

**Reply by mashb1t** [reply]
... which is also why i've abandoned it, see https://github.com/mashb1t/Fooocus/discussions/13

**Reply by tybiboune** [reply]
ah that's a shame :) nothing else but GRADIO out there? It seems to have many downsides (I assume it's easier to use?)

**Reply by mashb1t** [reply]
Gradio is the perfect tool for a project like Fooocus or A1111 as it uses websockets by default and has a queue for multiuser access. This is in contrast with a single-user speedy 1:1 only interface, which e.g. StreamDiffusion offers. IMHO live image generation with lower quality is better suited to be in its separate tool and this project should fo(oo)cus on the things where it's really good at, mainly high quality image generation and implementation of technology innovations in an easy-to-use interface.

**Comment by tybiboune** [comment]
well thanks for this precise explanation, really makes sense!

Le mar. 13 févr. 2024 à 16:03, Manuel Schmid ***@***.***> a
écrit :
